### DISCUSSION TITLE
Opt out Popover close on Enter key behavior

### DISCUSSION THREAD

**Discussion by nirtamir2** [radix-ui/primitives#1685] [Help] [answered]
I have a form element inside a popover.
When I open the popover and fill out the form I tend to press Enter key in order to submit the form.
But the popover behavior catches this Enter key and closes the form instead of submitting the form.

After searching in the docs https://www.radix-ui.com/docs/primitives/components/popover#keyboard-interactions I notice this is the expected popover behavior to close on enter
![image](https://user-images.githubusercontent.com/16452789/192132019-08be88f7-9092-4df7-b9c8-c7e4e550554f.png)

I end up doing something like:
```
<form
      // eslint-disable-next-line @typescript-eslint/no-misused-promises
      onSubmit={handleSubmit(onSubmit)}
      onKeyDown={(e) => {
        if (e.key === "Enter") {
          e.stopPropagation();
          void handleSubmit(onSubmit)(e);
        }
      }}
    />
```

- Why does the popover element close with the `Enter` key?
- Can we opt out of this behavior?
- Should I handle it differently? What is the best practice in this case?

Thank!

**Comment by benoitgrelard** [comment]
Hi @nirtamir2,

> Why does the popover element close with the Enter key?

I don't think it does. If you look at the demo on the docs, it doesn't do that.
What we are referring to is pressing enter on the trigger/close button. That will close the Popover.

If you are seeing something else, could you link up a codesandbox with your issue?

**Comment by nirtamir2** [comment, marked as answer]
@benoitgrelard thank you for your help.
I fixed the problem - it has nothing related to radix-ui.
If you are interested - I have a form with cancel button and submit button and the cancel button did not have a type attribute.
So when I clicked Enter key on the input - the browser clicked the cancel button it instead of the submit button so it closed the popover. I added a reproduction here [codesandbox.io/s/frosty-tristan-f98fud?file=/src/App.tsx](https://codesandbox.io/s/frosty-tristan-f98fud?file=/src/App.tsx) with the problem and solution (form with button without type attribute).

**Reply by benoitgrelard** [reply]
That would make sense, because `button` without `type` inside a `form` become `type="submit"`.
